src/Reflection/Exception/NotAClassReflection.php

```diff
@@ -5,17 +5,25 @@
 namespace Roave\BetterReflection\Reflection\Exception;
 
 use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflector\Exception\IdentifierNotFound;
 use UnexpectedValueException;
 use function sprintf;
 
 class NotAClassReflection extends UnexpectedValueException
 {
     public static function fromReflectionClass(ReflectionClass $class) : self
     {
-        $type = 'interface';
-
         if ($class->isTrait()) {
             $type = 'trait';
+        } elseif ($class->isInterface()) {
+            $type = 'interface';
+        } else {
+            $type = 'unknown';
+            try {
+                $class->getParentClass();
+            } catch (IdentifierNotFound $e) {
```

Comment:
Doing this is a bit risky. Is there a way to avoid the operation at all, even if the exception message is a bit worse?

Reply:
I run `vendor\roave\backward-compatibility-check\bin\roave-backward-compatibility-check.php` with xdebug (and with `-dmemory_limit=2G`) and realized that you are right. Not found class became interface. So, there won't be any "unknown" and "try-catch" blocks. It is that issue 77. I will close PR